tests/apps/search/test_elasticsearch_compat_layer.py

```diff
@@ -22,48 +47,77 @@ class ElasticSearchCompatLayerTestCase(TestCase):
     different versions of ElasticSearch.
     """
 
-    @mock.patch(
-        "elasticsearch.Elasticsearch.info",
-        return_value={"should not be called": "the value is cached"},
+    @mock.patch.object(
+        CategoriesIndexer,
+        "index_name",
+        new_callable=mock.PropertyMock,
+        return_value="test_categories",
     )
-    def test_cache_version_information(self, mock_es_info):
+    def test_cache_version_information(self, *_):
         """
-        Make sure we only ever make the info() call once even if we use more than
-        one ElasticSearch API endpoint.
+        Make sure we only ever make the info() call once by instance
+        even if we use more than one ElasticSearch API endpoint.
         """
 
-        # Perform a bunch of ES actions, from index management all the way to searches
-        ES_INDICES_CLIENT.delete(index="_all")
-        ES_INDICES_CLIENT.create(index="test_categories")
-        ES_INDICES_CLIENT.close(index="test_categories")
-        ES_INDICES_CLIENT.put_settings(body=ANALYSIS_SETTINGS, index="test_categories")
-        ES_INDICES_CLIENT.open(index="test_categories")
-        ES_INDICES_CLIENT.put_mapping(
-            body=CategoriesIndexer.mapping, index="test_categories"
+        # - Use a fresh ES client instance to be sure that __es_version__ has
+        #   not been called yet
+        es_client = ElasticsearchClientCompat7to6(
+            [getattr(settings, "RICHIE_ES_HOST", "elasticsearch")]
         )
+        es_indices_client = ElasticsearchIndicesClientCompat7to6(es_client)
 
-        actions = [
-            {
-                "_id": category["id"],
-                "_index": "test_categories",
-                "_op_type": "create",
-                "absolute_url": {"en": "en/url"},
-                "description": {"en": "en/description"},
-                "icon": {"en": "en/icon"},
-                "is_meta": False,
-                "logo": {"en": "en/logo"},
-                "nb_children": 0,
-                "path": category["id"],
-                "title_raw": category["title"],
-                **category,
-            }
-            for category in CATEGORIES
-        ]
-        bulk_compat(actions=actions, chunk_size=500, client=ES_CLIENT)
-        ES_INDICES_CLIENT.refresh()
+        with mock.patch(
+            "elasticsearch.Elasticsearch.info", wraps=es_client.info
+        ) as mock_es_info:
+            with mock.patch("richie.apps.search.ES_CLIENT", es_client):
+                with mock.patch(
+                    "richie.apps.search.ES_INDICES_CLIENT", es_indices_client
+                ):
```

Comment:
You can avoid indentation:
```python
with mock.patch("elasticsearch.Elasticsearch.info") as mock_info, mock.patch(
    "richie.apps.search.ES_CLIENT"
) as mock_es_info, mock.patch(
    "richie.apps.search.ES_INDICES_CLIENT"
) as mock_es_indices_client:
    # Perform a bunch of ES actions, from index management all the way to searches
```

Actually you only need to grab `mock_es_info` so:
```python
with mock.patch("elasticsearch.Elasticsearch.info"), mock.patch(
    "richie.apps.search.ES_INDICES_CLIENT"
), mock.patch("richie.apps.search.ES_CLIENT") as mock_es_info:
    # Perform a bunch of ES actions, from index management all the way to searches
```

Not why you mock the two other things but I will trust you on this :stuck_out_tongue:

Reply:
Because If I do not use a mock here, `ES_CLIENT` and `ES_INDICES_CLIENT` can already be instanciated before and assertion `mock_es_info.assert_called_once()` fails.